### PR TITLE
Add instruction about release branches

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+2.  For `master` branch:
+
+    -   Change version files to the next minor snapshot (e.g. `0.2.0-dev`).
+
+    ```bash
+    $ git checkout -b bump-version master
+    # Change version to next minor (and keep -dev)
+    $ sed -i 's/[0-9]\+\.[0-9]\+\.[0-9]\+\(.*CURRENT_OPENCENSUS_VERSION\)/'$MAJOR.$((MINOR+1)).0'\1/' \
+      "${VERSION_FILES[@]}"
+    # // TODO: Add instructions how to check that everything passes tests.
```
Comment (g-easy) [inline RELEASING.md]: Please take a look at `tools/presubmit.sh`

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    ```bash
+    # Change version to next patch and add -dev
+    $ sed -i 's/[0-9]\+\.[0-9]\+\.[0-9]\+\(.*CURRENT_OPENCENSUS_VERSION\)/'$MAJOR.$MINOR.$((PATCH+1))-dev'\1/' \
+     "${VERSION_FILES[@]}"
+    # // TODO: Add instructions how to check that everything passes tests.
+    $ git commit -a -m "Bump version to $MAJOR.$MINOR.$((PATCH+1))-dev"
+    ```
+
+    -   Go through PR review and push the release tag and updated release branch
```
Comment (g-easy) [inline RELEASING.md]: Do we still need to https://github.com/census-instrumentation/opencensus-cpp/releases and "Draft a new release" ?

File: RELEASING.md (hunk 1 above)
Comment (bogdandrutu) [reply RELEASING.md]: Done.

File: RELEASING.md (hunk 2 above)
Comment (bogdandrutu) [reply RELEASING.md]: Done.

**Comment by bogdandrutu** [comment]
PTAL?

**Pull request merged**